lib/statsd/instrument/client.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'statsd/instrument/datagram'
+require 'statsd/instrument/datagram_builder'
+require 'statsd/instrument/statsd_datagram_builder'
+require 'statsd/instrument/dogstatsd_datagram_builder'
+require 'statsd/instrument/null_sink'
+require 'statsd/instrument/udp_sink'
+require 'statsd/instrument/capture_sink'
+require 'statsd/instrument/log_sink'
+
+class StatsD::Instrument::Client
+  attr_reader :sink, :datagram_builder_class, :prefix, :default_tags, :sampling_disabled, :default_sample_rate
+
+  def initialize(
+    sink: StatsD::Instrument::NullSink.new,
+    prefix: nil,
+    sampling_disabled: false,
```

Comment:
I wasn't too sure what the best way to handle it.

If we leave it up to the sink, it needs to know the sample rate of the datagram. It can do that by parsing the string (a no go area), or we have to explicitly pass it in which requires changing the contract. I kind of liked that right now a sink only needs to implement `<<(packet)` and we can mock it with a simple array. But the alternatives are probably worse.